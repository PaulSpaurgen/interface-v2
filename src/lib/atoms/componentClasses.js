const baseFilledButtonClasses = 'btn btn-theme btn-sm items-center flex justify-center';
export const buttonClasses = {
    filled: `${baseFilledButtonClasses} btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white`,
    greyFilled: `${baseFilledButtonClasses} bg-base-200 text-primary border-none hover:bg-base-300 `,
    lightblueFilled: `${baseFilledButtonClasses} btn-lightblue`,
    whiteFilled: `${baseFilledButtonClasses} bg-white text-primary border-none hover:bg-base-200`,
    outlined: 'btn btn-theme btn-sm text-primary px-4 border-primary bg-inherit hover:text-primary-focus hover:border-primary-focus hover:bg-inherit disabled:bg-inherit disabled:text-grey-300 disabled:border-grey-300',
    error: 'btn btn-theme btn-sm text-error px-5 border-error bg-inherit hover:bg-inherit hover:border-error hover:text-error-focus',
    text: 'btn btn-sm btn-ghost p-0 bg-transparent hover:bg-transparent',
    icon: 'btn btn-xs btn-ghost btn-circle p-0 bg-transparent hover:bg-transparent flex',
    maxButton: 'btn btn-sm btn-text btn-ghost p-0 bg-transparent hover:bg-transparent font-bold text-primary tracking-widest disabled:text-primary',
    largeButton: 'flex gap-2 text-primary h-14 text-base font-semibold items-center',
    tableConvertButton: 'btn btn-xs text-xs h-[35px] w-[126px] rounded btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white pt-1'
};
export const dividerClasses = {
    vertical: 'w-[1px] bg-gray-500 h-4 rounded',
    horizontal: 'h-0.5 bg-gray-200 w-full rounded',
    verticalPrimary: 'w-[1px] bg-primary h-4 rounded'
};
export const tableCellClasses = {
    heading: 'text-primary text-sm font-semibold px-2 pb-4',
    row: 'font-semibold text-base px-2 py-3',
    rowMini: 'font-normal text-xs px-1 py-1',
    rowWithIcon: 'flex flex-col items-center justify-start',
    empty: 'text-base text-center p-4 w-full'
};
//# sourceMappingURL=componentClasses.js.map