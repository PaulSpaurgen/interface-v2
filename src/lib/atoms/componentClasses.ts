const baseFilledButtonClasses = 'btn btn-theme btn-sm px-5 items-center';

export const buttonClasses = {
	filled: `${baseFilledButtonClasses} btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white`,
	greyFilled: `${baseFilledButtonClasses} bg-base-200 text-primary border-none hover:bg-base-300`,
	lightblueFilled: `${baseFilledButtonClasses} btn-lightblue`,
	whiteFilled: `${baseFilledButtonClasses} bg-white text-primary border-none hover:bg-base-200`,
	outlined:
		'btn btn-theme btn-sm text-primary px-4 border-primary bg-inherit hover:text-primary-focus hover:border-primary-focus hover:bg-inherit',
	error:
		'btn btn-theme btn-sm text-error px-5 border-error bg-inherit hover:bg-inherit hover:border-error hover:text-error-focus',
	text: 'btn btn-sm btn-ghost p-0 bg-transparent hover:bg-transparent',
	icon: 'btn btn-xs btn-ghost btn-circle p-0 bg-transparent hover:bg-transparent flex',
	maxButton:
		'btn btn-sm btn-ghost p-0 bg-transparent hover:bg-transparent text-sm font-bold text-primary',
	largeButton: 'flex gap-2 text-primary h-14 text-base font-semibold items-center',
	convertButton: 'btn btn-xs text-xs h-4 py-0'
};

export const dividerClasses = {
	vertical: 'w-[1px] bg-gray-500 h-4 rounded',
	horizontal: 'h-0.5 bg-gray-200 w-full rounded',
	verticalPrimary: 'w-[1px] bg-primary h-4 rounded'
};

export const tableCellClasses = {
	heading: 'text-primary text-sm font-semibold px-2 py-6',
	row: 'font-semibold px-2 py-4',
	rowWithIcon: 'flex flex-col items-center justify-start',
	empty: 'text-base text-center p-4 w-full'
};
