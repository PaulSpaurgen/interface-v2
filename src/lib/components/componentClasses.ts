export const buttonClasses = {
	primary: 'btn btn-sm text-sm rounded btn-primary',
	secondary:
		'btn btn-sm text-sm rounded text-primary border-primary hover:text-primary-focus hover:border-primary-focus hover:bg-inherit',
	error: 'btn btn-sm rounded text-error border-error hover:bg-white hover:border-error',
	icon: 'btn btn-ghost btn-sm btn-circle'
};
