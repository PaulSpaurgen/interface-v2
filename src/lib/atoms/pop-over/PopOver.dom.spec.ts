import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/svelte';
import Popover from './PopOver.svelte';
import { buttonClasses } from '../componentClasses';


describe('YourComponent', () => {
    it('should render popover correctly', () => {
        const { container } = render(Popover);
        expect(container).toMatchSnapshot();

    })
    it('should render a button with the correct class from buttonClasses', () => {
        const { getByTestId } = render(Popover);
        expect(getByTestId('popover-button').className).contain(buttonClasses.icon)
    });
});
