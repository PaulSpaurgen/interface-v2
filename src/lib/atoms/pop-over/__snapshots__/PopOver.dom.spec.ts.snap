// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YourComponent > should render a button with the correct class from buttonClasses 1`] = `
<body>
  <div>
    <div
      class="dropdown dropdown-top dropdown-hover"
    >
      <button
        class="btn btn-xs btn-ghost btn-circle p-0 bg-transparent hover:bg-transparent flex disabled:cursor-not-allowed"
        data-testid="popover-button"
        tabindex="0"
      />
       
      <div
        class="menu dropdown-content w-64 rounded bg-base-100 shadow"
      />
    </div>
    <!--&lt;PopOver&gt;-->
  </div>
</body>
`;

exports[`YourComponent > should render popover correctly 1`] = `
<body>
  <div>
    <div
      class="dropdown dropdown-top dropdown-hover"
    >
      <button
        class="btn btn-xs btn-ghost btn-circle p-0 bg-transparent hover:bg-transparent flex disabled:cursor-not-allowed"
        data-testid="popover-button"
        tabindex="0"
      />
       
      <div
        class="menu dropdown-content w-64 rounded bg-base-100 shadow"
      />
    </div>
    <!--&lt;PopOver&gt;-->
  </div>
</body>
`;
